lib/web/fetch/index.js

```diff
@@ -133,116 +178,147 @@ function fetch (input, init = undefined) {
-      //    relevantRealm.
+    // 13. Set controller to the result of calling fetch given request,
+    // with processResponseEndOfBody set to handleFetchDone, and processResponse
+    // given response being these substeps:
 
-      // 2. Abort the fetch() call with p, request, responseObject, and
-      //    deserializedError.
+    const processResponse = (response) => {
+      // 1. If locallyAborted is true, terminate these substeps.
+      if (locallyAborted) {
+        return Promise.resolve()
+      }
 
-      abortFetch(p, request, responseObject, controller.serializedAbortReason)
-      return
-    }
+      try {
+        // 2. If response’s aborted flag is set, then:
+        if (response.aborted) {
+          // 1. Let deserializedError be the result of deserialize a serialized
+          //    abort reason given controller’s serialized abort reason and
+          //    relevantRealm.
 
-    // 3. If response is a network error, then reject p with a TypeError
-    // and terminate these substeps.
-    if (response.type === 'error') {
-      p.reject(new TypeError('fetch failed', { cause: response.error }))
-      return
-    }
+          // 2. Abort the fetch() call with p, request, responseObject, and
+          //    deserializedError.
 
-    // 4. Set responseObject to the result of creating a Response object,
-    // given response, "immutable", and relevantRealm.
-    responseObject = fromInnerResponse(response, 'immutable', relevantRealm)
+          abortFetch(p, request, responseObject, controller.serializedAbortReason, context)
+          return Promise.resolve()
+        }
 
-    // 5. Resolve p with responseObject.
-    p.resolve(responseObject)
-  }
+        // 3. If response is a network error, then reject p with a TypeError
+        // and terminate these substeps.
+        if (response.type === 'error') {
+          publishIfContext('fetchError', context, response.error)
+          p.reject(new TypeError('fetch failed', { cause: response.error }))
+          return Promise.resolve()
+        }
 
-  controller = fetching({
-    request,
-    processResponseEndOfBody: handleFetchDone,
-    processResponse,
-    dispatcher: requestObject[kDispatcher] // undici
-  })
+        // 4. Set responseObject to the result of creating a Response object,
+        // given response, "immutable", and relevantRealm.
+        responseObject = fromInnerResponse(response, 'immutable', relevantRealm)
+
+        if (hasSubscribers) {
+          context.result = responseObject[kState]
+        }
 
+        // 5. Resolve p with responseObject.
+      } finally {
+        publishIfContext('asyncStart', context)
+        publishIfContext('asyncEnd', context)
+        p.resolve(responseObject)
+      }
+    }
+
+    controller = fetching({
+      request,
+      processResponseEndOfBody: handleFetchDone,
+      processResponse,
+      dispatcher: init?.dispatcher ?? getGlobalDispatcher() // undici
```

Comment:
Is this change intentional? Doesn't seem related. 🤔

Reply:
fixed!